_episodes/24.md

```diff
@@ -0,0 +1,412 @@
+    query.max-memory=4GB
+    query.max-memory-per-node=1GB
+    query.max-total-memory-per-node=2GB
+    memory.heap-headroom-per-node=1GB
+    discovery-server.enabled=true
+    discovery.uri=http://localhost:8080
+
+  log.properties: |
+    io.trino=INFO
+EOF
+```
+
+Add the tcb-trino service definition to run Trino. 
+```
+kubectl apply -f - <<EOF
+# Source: trino/templates/service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: tcb-trino
+  labels:
+    app: trino
+    chart: trino-0.2.0
+    release: tcb
+    heritage: Helm
+spec:
+  type: ClusterIP
+  ports:
+    - port: 8080
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    app: trino
+    release: tcb
+    component: coordinator
+EOF
+```
+
+Add the deployment definition for the service.
+```
+kubectl apply -f - <<EOF
+# Source: trino/templates/deployment-coordinator.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tcb-trino-coordinator
+  labels:
+    app: trino
+    chart: trino-0.2.0
+    release: tcb
+    heritage: Helm
+    component: coordinator
+spec:
+  selector:
+    matchLabels:
+      app: trino
+      release: tcb
+      component: coordinator
+  template:
+    metadata:
+      labels:
+        app: trino
+        release: tcb
+        component: coordinator
+    spec:
+      securityContext:
+        runAsUser: 1000
+        runAsGroup: 1000
+      volumes:
+        - name: config-volume
+          configMap:
+            name: tcb-trino-coordinator
+        - name: catalog-volume
+          configMap:
+            name: tcb-trino-catalog
+      imagePullSecrets:
+        - name: registry-credentials
+      containers:
+        - name: trino-coordinator
+          image: "trinodb/trino:latest"
+          imagePullPolicy: IfNotPresent
+          volumeMounts:
+            - mountPath: /etc/trino
+              name: config-volume
+            - mountPath: /etc/trino/catalog
+              name: catalog-volume
+          ports:
+            - name: http
+              containerPort: 8080
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /v1/info
+              port: http
+          readinessProbe:
+            httpGet:
+              path: /v1/info
+              port: http
+          resources:
+            {}
+EOF
+```
+
+Now check the state of the k8s cluster again.
+```
+kubectl get all
+```
+
+Run the following command to expose the url and port to the localhost system.
+```
+minikube service tcb-trino --url
+```
+
+Clean up all the resources.
+```
+kubectl delete pod --all
+kubectl delete replicaset --all
+kubectl delete service tcb-trino
+kubectl delete deployment tcb-trino-coordinator
+kubectl delete configmap --all
+```
+
+Now you can run the same demo using the helm chart which includes all of these
+templates out-of-the-box. First add the trino helm chart, check the templates
+that are produced by helm, and run the install.
+```
+# HELM DEMO
```

Comment:
Is this heading suposed to be inside the code block or an actual heading?

Reply:
That's okay this was just to split it in the code portion of the demo but wasn't intended for markdown.